commcare_connect/utils/ocs_api.py

```
@@ -0,0 +1,75 @@
+def list_chatbots(user) -> list[tuple[str, str]]:
+    """Return ``[(id, name), ...]`` for every OCS chatbot, following cursor pagination."""
+    token = _valid_token(user)
+    headers = {"Authorization": f"Bearer {token.token}"}
+    url = f"{settings.OCS_BASE_URL}/api/v2/chatbots/"
+    chatbots = []
+    while url:
+        try:
+            response = httpx.get(url, headers=headers)
+            response.raise_for_status()
+        except httpx.HTTPStatusError:
+            raise OcsApiError(f"Failed to list chatbots: {response.text}")
```

Ruff (0.15.20)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)